ethportal-api/src/types/network.rs

```diff
@@ -29,32 +33,109 @@ impl std::str::FromStr for Network {
+            Subnetwork::VerkleState => write!(f, "Verkle State"),
+            Subnetwork::TransactionGossip => write!(f, "Transaction Gossip"),
+            Subnetwork::Utp => write!(f, "uTP"),
         }
     }
 }
 
-impl std::str::FromStr for Subnetwork {
-    type Err = String;
-
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
+impl Subnetwork {
+    pub fn from_cli_arg(s: &str) -> Result<Self, String> {
         match s {
             "beacon" => Ok(Subnetwork::Beacon),
             "history" => Ok(Subnetwork::History),
             "state" => Ok(Subnetwork::State),
+            "canonical_indices" => Ok(Subnetwork::CanonicalIndices),
+            "verkle_state" => Ok(Subnetwork::VerkleState),
+            "transaction_gossip" => Ok(Subnetwork::TransactionGossip),
+            "utp" => Ok(Subnetwork::Utp),
             _ => Err(format!("Unknown subnetwork: {s}")),
         }
     }
+
+    pub fn to_cli_arg(&self) -> String {
+        match self {
+            Subnetwork::Beacon => "beacon".to_string(),
+            Subnetwork::History => "history".to_string(),
+            Subnetwork::State => "state".to_string(),
+            Subnetwork::CanonicalIndices => "canonical_indices".to_string(),
+            Subnetwork::VerkleState => "verkle_state".to_string(),
+            Subnetwork::TransactionGossip => "transaction_gossip".to_string(),
+            Subnetwork::Utp => "utp".to_string(),
+        }
+    }
+}
+
+impl From<Subnetwork> for u8 {
+    fn from(subnetwork: Subnetwork) -> u8 {
+        match subnetwork {
+            Subnetwork::Beacon => 1,
+            Subnetwork::History => 0,
+            Subnetwork::State => 2,
+            Subnetwork::CanonicalIndices => 4,
+            Subnetwork::VerkleState => 5,
+            Subnetwork::TransactionGossip => 3,
+            Subnetwork::Utp => 99,
+        }
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum SubnetworkError {
+    #[error("Unable to decode subnetwork to bytes")]
+    Decode(ByteUtilsError),
+
+    #[error("invalid subnetwork")]
+    Invalid,
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn subnetwork_to_u8() {
+        let protocol_id = Subnetwork::History;
+        let expected_u8 = 0;
+        assert_eq!(expected_u8, u8::from(protocol_id));
+
+        let protocol_id = Subnetwork::Beacon;
+        let expected_u8 = 1;
+        assert_eq!(expected_u8, u8::from(protocol_id));
+
+        let protocol_id = Subnetwork::State;
+        let expected_u8 = 2;
+        assert_eq!(expected_u8, u8::from(protocol_id));
+
+        let protocol_id = Subnetwork::TransactionGossip;
+        let expected_u8 = 3;
+        assert_eq!(expected_u8, u8::from(protocol_id));
+
+        let protocol_id = Subnetwork::CanonicalIndices;
+        let expected_u8 = 4;
+        assert_eq!(expected_u8, u8::from(protocol_id));
+
+        let protocol_id = Subnetwork::Utp;
+        let expected_u8 = 99;
+        assert_eq!(expected_u8, u8::from(protocol_id));
```

Comment:
These aren't protocol_identifier's. We are just mimicking an enum as an identifier for the database with this code no?